File: 09-DesignPatterns-Java/Solution/State/src/States/AngryState.java
package States;

public final class AngryState extends PartnerStateBase {

    public AngryState(Partner.Partner partner) {
        super(partner);
    }

    @Override
    public void talk() {
        print("Go away, I am angry!");
    }

    @Override
    public void kiss() {
        print("Okay, I forgive you.");
        setState(new NeutralState(_partner));
    }

    @Override
    public void insult() {
        print("You shall burn in hell.");
        setState(new SuperAngryState(_partner));
    }
}

File: 10-DesignPatterns-CSharp/Solution/State/States/AngryState.cs
namespace State.States
{
    internal sealed class AngryState : PartnerStateBase
    {
        public AngryState(Partner partner)
            : base(partner)
        {
        }

        public override void Talk()
        {
            Print("Go away, I am angry!");
        }

        public override void Kiss()
        {
            Print("Okay, I forgive you.");
            SetState(new NeutralState(Partner));
        }

        public override void Insult()
        {
            Print("You shall burn in hell!");
            SetState(new SuperAngryState(Partner));
        }
    }
}
